junit-jupiter-api/junit-jupiter-api.gradle.kts

```diff
@@ -14,4 +14,11 @@ dependencies {
 	api(project(":junit-platform-commons"))
 
 	compileOnly("org.jetbrains.kotlin:kotlin-stdlib")
+
+	testImplementation("org.jetbrains.kotlin:kotlin-stdlib")
+	testRuntimeOnly(project(":junit-jupiter-engine"))
```

Comment:
The `junit-jupiter-api` project is not allowed to contain tests, since that introduces a classpath cycle in certain IDEs (e.g., Eclipse).

All tests for `junit-jupiter-api` reside in `junit-jupiter-engine`, so please undo this change and move the tests to `junit-jupiter-engine`.

Reply:
I assumed as much and that's also why I asked for feedback regarding this. That said, these are not really tests but examples and they must be accessible for the `junit-jupiter-api` module or they cannot be referenced in the documentation blocks. I see a few possible ways to go about this:

1. Live with the fact that the examples might get outdated or are outright wrong because there is no validation (status quo).
1. Make the test classpath of `junit-jupiter-engine` available at compile time to `junit-jupiter-api` (does not really resolve the cycle issue).
1. Do what I have done here which works because `compile` happens before `compileTest` and there is thus no real cycle.
1. Instead of using `prject(":junit-jupiter-engine")` for testing of the samples we use a previously released version of JUnit. This should not be a problem because the engine does not provide anything on top of `junit-jupiter-api` that is required in the samples and which would lead to issues since the samples target only `junit-jupiter-api`.

Reply:
How about we put them into the `testFixtures` and add a test to `platform-tests` that runs them?

Reply:
That sounds good. 👍